### PR TITLE
Update TLS Config & Add ListenMutualTLS

### PR DESCRIPTION
**Pull request by efectn**
- Remove deprecations on tls.Config.
- Add ListenMutualTLS method.

**Closes:** https://github.com/gofiber/fiber/issues/1715

### CODE REVIEW

**Comment by ReneWerner87** [comment]
- [ ] Don't forget the documentation repository

**Review by ReneWerner87 — APPROVED** [review]

File: app.go [hunk 1]
```diff
@@ -814,8 +817,7 @@ func (app *App) ListenTLS(addr, certFile, keyFile string) error {
 			return fmt.Errorf("tls: cannot load TLS key pair from certFile=%q and keyFile=%q: %s", certFile, keyFile, err)
 		}
 		config := &tls.Config{
-			MinVersion:               tls.VersionTLS12,
-			PreferServerCipherSuites: true,
```
Comment (hi019) [inline app.go:818]: would this be a breaking change?

File: app.go [hunk 2]
```diff
@@ -841,6 +843,66 @@ func (app *App) ListenTLS(addr, certFile, keyFile string) error {
 	return app.server.ServeTLS(ln, certFile, keyFile)
 }
 
+// ListenMutualTLS serves HTTPs requests from the given addr.
+// certFile, keyFile and clientCertFile are the paths to TLS certificate and key file.
+
+//  app.ListenMutualTLS(":8080", "./cert.pem", "./cert.key", "./client.pem")
+//  app.ListenMutualTLS(":8080", "./cert.pem", "./cert.key", "./client.pem")
```
Comment (hi019) [inline app.go:850]:
```suggestion
//  app.ListenMutualTLS(":8080", "./cert.pem", "./cert.key", "./client.pem")
```

File: app.go [hunk 3]
```diff
@@ -814,8 +817,7 @@ func (app *App) ListenTLS(addr, certFile, keyFile string) error {
 			return fmt.Errorf("tls: cannot load TLS key pair from certFile=%q and keyFile=%q: %s", certFile, keyFile, err)
 		}
 		config := &tls.Config{
-			MinVersion:               tls.VersionTLS12,
-			PreferServerCipherSuites: true,
+			MinVersion: tls.VersionTLS13,
```
Comment (hi019) [inline app.go:820]: would this be a breaking change?

File: app.go (hunk 2 above)
Comment (hi019) [inline app.go:850]:
```suggestion
//  app.ListenMutualTLS(":8080", "./cert.pem", "./cert.key", "./client.pem")
```

**Review by hi019 — CHANGES_REQUESTED** [review]

File: app.go (hunk 3 above)
Comment (efectn) [reply app.go:820]: I'm not sure but it would be breaking change for old certficates may be.

Comment (balcieren) [reply app.go:820]: `MinVersion: tls.VersionTLS12`
`MaxVersion: tls.VersionTLS13`

can it prevent breaking change ?

Comment (efectn) [reply app.go:820]: I think we dont need it. Also i've seen https://tip.golang.org/doc/go1.18#tls10. I think we can use tls1.2 as minimum version instead. I can revert it. @ReneWerner87 @hi019

**Comment by ReneWerner87** [comment]
@hi019 please check again

**Comment by hi019** [comment]
Looks good. I'll test it out today or tomorrow

**Review by hi019 — APPROVED** [review]

**Pull request merged**